python/cudf/cudf/core/dataframe.py

```diff
@@ -6169,23 +5892,15 @@ def append(
             other = other.reindex(combined_columns, copy=False).to_frame().T
             if not current_cols.equals(combined_columns):
                 self = self.reindex(columns=combined_columns)
-        elif isinstance(other, list):
-            if not other:
-                pass
-            elif not isinstance(other[0], DataFrame):
-                other = DataFrame(other)
-                cols = self._data.to_pandas_index()
-                if (
-                    cols.get_indexer(other._data.to_pandas_index()) >= 0
-                ).all():
-                    other = other.reindex(columns=cols)
-
-        if is_list_like(other):
-            to_concat = [self, *other]
-        else:
-            to_concat = [self, other]
+        elif isinstance(other, list) and not isinstance(other[0], DataFrame):
```

Comment:
```suggestion
        elif isinstance(other, list) and len(other) > 1 and not isinstance(other[0], DataFrame):
```
This check could probably fail if `other` is `[]`

Reply:
Good catch.